### PR TITLE
Fix ARPAgeCache function CBMC proof

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
Description
-----------

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
+
+/* Provide a stub for the interface output function. This is the low
+ * level output function which is platform dependent. */
+BaseType_t xLocalNetworkInterfaceOutput( NetworkInterface_t * pxInterface,
+                                         NetworkBufferDescriptor_t * const pxNetworkBuffer,
+                                         BaseType_t bReleaseAfterSend )
+{
+    BaseType_t xReturn;
+
+    return xReturn;
```
Comment (markrtuttle) [inline test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c:20]: What about checking (asserting) that the pointer arguments are nonnull?

File: test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c [hunk 2]
```diff
@@ -25,5 +38,43 @@ NetworkBufferDescriptor_t * pxGetNetworkBufferWithDescriptor( size_t xRequestedS
...
+
+    const uint8_t ucIPAddress2[ 4 ];
+    const uint8_t ucNetMask2[ 4 ];
+    const uint8_t ucGatewayAddress2[ 4 ];
+    const uint8_t ucDNSServerAddress2[ 4 ];
+    const uint8_t ucMACAddress[ 6 ];
+
+    /* Assume that the list of interfaces/endpoints is not initialized. */
+    __CPROVER_assume( pxNetworkInterfaces == NULL );
+    __CPROVER_assume( pxNetworkEndPoints == NULL );
```
Comment (markrtuttle) [inline test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c:52]: Are these two variables even defined?

**Review by markrtuttle — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c (hunk 2 above)
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c:52]: They are defined in one of the included files. They are defined as NULL there. Just copied the assumes here as well :)

File: test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c (hunk 1 above)
Comment (AniruddhaKanhere) [reply test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c:20]: Good catch! I'll add it.

Comment (AniruddhaKanhere) [reply test/cbmc/proofs/ARP/ARPAgeCache/ARPAgeCache_harness.c:25]: Added.

**Review by markrtuttle — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**